src/lib/rng/auto_rng/auto_rng.cpp

```diff
@@ -6,19 +6,22 @@
 
 #include <botan/auto_rng.h>
 
-#include <botan/entropy_src.h>
 #include <botan/hmac_drbg.h>
 #include <botan/internal/loadstor.h>
 #include <botan/internal/os_utils.h>
 
-#include <array>
+#if defined(BOTAN_HAS_ENTROPY_SOURCE)
+   #include <botan/entropy_src.h>
+#endif
 
 #if defined(BOTAN_HAS_SYSTEM_RNG)
    #include <botan/system_rng.h>
 #endif
 
 namespace Botan {
 
+class Entropy_Sources;
+
```

Comment:
This relates to lines 68 to 72 where `Entropy_Sources` is actually used of Botan doesn't have the system rng enabled. Probably, this needs another ifdef.

Reply:
Wasn't even required at al - removed

Reply:
Did you push all changes here? If I configure without `entropy` and `system_rng` botan fails to build with:

```
src/lib/rng/auto_rng/auto_rng.cpp:68:22: error: incomplete type 'Botan::Entropy_Sources' named in nested name specifier
   68 |       AutoSeeded_RNG(Entropy_Sources::global_sources(), reseed_interval)
      |                      ^~~~~~~~~~~~~~~~~
build/include/public/botan/rng.h:26:7: note: forward declaration of 'Botan::Entropy_Sources'
   26 | class Entropy_Sources;
      |       ^
```

Reply:
Missing add, fixed now